GP11 : operation state handling

cedric-chaumont-st-dev opened the pull request:
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

cedric-chaumont-st-dev commented:
Rebased.

jforissier commented on lib/libutee/tee_api_operations.c:
```diff
@@ -810,17 +826,30 @@ TEE_Result TEE_DigestDoFinal(TEE_OperationHandle operation, const void *chunk,
+	if (operation->operationState != TEE_OPERATION_STATE_INITIAL)
+		TEE_ResetOperation(operation);
+
+	if (operation->operationState == TEE_OPERATION_STATE_INITIAL)
```
This `if ()` is not needed because `TEE_ResetOperation()` always sets `operationState` to `INITIAL` (or panics).

jforissier commented on lib/libutee/tee_api_operations.c:
```diff
@@ -1081,10 +1123,20 @@ void TEE_MACInit(TEE_OperationHandle operation, void *IV, uint32_t IVLen)
+	if (operation->operationState != TEE_OPERATION_STATE_INITIAL)
+		TEE_ResetOperation(operation);
+
+	if (operation->operationState == TEE_OPERATION_STATE_INITIAL)
```
Same as above.

jforissier commented:
`Reviewed-by: Jerome Forisssier <jerome.forissier@linaro.org>`

pascal-brand-st-dev commented:
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

cedric-chaumont-st-dev commented:
Updated. Rebased. Verified (STM, ARm Juno).
+-----------------------------------------------------
29360 subtests of which 0 failed
687 test cases of which 0 failed
0 test case was skipped
TEE test application done!
